.github/workflows/test_e2eshark.yml

```diff
@@ -0,0 +1,175 @@
+        os: [nodai-amdgpu-w7900-x86-64]
+    runs-on: ${{matrix.os}}
+    # steps:
+    #   - name: Checkout repo
+    #     uses: actions/checkout@v2
+    #     with:
+    #       submodules: true
+    #       repository: llvm/torch-mlir
+    #       path: torch-mlir
+      
+    #   - name: "Setting up Python"
+    #     uses: actions/setup-python@v5
+    #     with:
+    #       python-version: ${{matrix.version}}
+
+    #   - name: Build torch-mlir
+    #     run: |
+    #       git submodule update --init --progres
+    #       python -m venv mlir_venv
+    #       source mlir_venv/bin/activate
+    #       pip install --upgrade pip
+    #       pip install -r ./externals/llvm-project/mlir/python/requirements.txt
+    #       pip install -r requirements.txt
+    #       pip install -r torchvision-requirements.txt
+    #       pip install "pybind11[global]"
+    #       cmake -S "$repo_root/externals/llvm-project/llvm" -B "$build_dir" \
+    #         -GNinja \
+    #         -DCMAKE_BUILD_TYPE=Release \
+    #         -DPython3_FIND_VIRTUALENV=ONLY \
+    #         -DLLVM_ENABLE_PROJECTS=mlir \
+    #         -DLLVM_EXTERNAL_PROJECTS="torch-mlir" \
+    #         -DLLVM_EXTERNAL_TORCH_MLIR_SOURCE_DIR="$PWD" \
+    #         -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
+    #         -DLLVM_TARGETS_TO_BUILD=host
+    #       cmake --build build
+    #       CMAKE_GENERATOR=Ninja python setup.py bdist_wheel --dist-dir ./torch-mlir-wheel -v
+    #       pip uninstall torch-mlir
+    #       pip install torch-mlir-wheel/torch_mlir-0.0.1-cp311-cp311-linux_x86_64.whl
+    #       deactivate
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          submodules: true
+          repository: saienduri/torch-mlir-fork
+          path: torch-mlir
+
+      - name: Install python deps (nightly)
+        run: |
+          python -m venv mlir_venv
+          source mlir_venv/bin/activate
+          bash build_tools/ci/install_python_deps.sh nightly
+        working-directory: ./torch-mlir
+
+      - name: Build project
+        run: |
+          bash build_tools/ci/build_posix.sh
+          pip uninstall torch-mlir
+          pip install torch-mlir-wheel/torch_mlir-0.0.1-cp311-cp311-linux_x86_64.whl
+          deactivate
+        working-directory: ./torch-mlir
+  
+  iree:
+    strategy:
+      matrix:
+        version: [3.11]
+        os: [nodai-amdgpu-w7900-x86-64]
+    runs-on: ${{matrix.os}}
+    env:
+      IREE_VENV_DIR: ${{ github.workspace }}/iree_venv
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          repository: openxla/iree
+          path: iree
+      
+      - name: "Setting up Python"
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{matrix.version}}
+
+      - name: "Setup Python venv"
+        run: python -m venv ${IREE_VENV_DIR}
+
+      - name: Build iree
```

Comment:
This (and all the other build steps) should be replaced with a pip install. No need to checkout a repo with all the submodules, configure toolchains (CMake, ninja, etc.), and spent CI time rebuilding from source.

Reply:
Yeah, it was a mess getting the whole source build into CI. Should be able to just use the pip install.